src/govuk/common/closest-attribute-value.mjs

```diff
@@ -3,13 +3,21 @@ import '../vendor/polyfills/Element/prototype/closest.mjs'
 /**
  * Returns the value of the given attribute closest to the given element (including itself)
  *
- * @param {HTMLElement} $element - The element to start walking the DOM tree up
+ * @param {Element} $element - The element to start walking the DOM tree up
  * @param {string} attributeName - The name of the attribute
  * @returns {string | undefined} Attribute value
  */
 export function closestAttributeValue ($element, attributeName) {
-  var closestElementWithAttribute = $element.closest('[' + attributeName + ']')
-  if (closestElementWithAttribute) {
-    return closestElementWithAttribute.getAttribute(attributeName)
+  var $closestElement = $element.closest('[' + attributeName + ']')
+  var closestAttributeValue
```

Comment:
Welcome to the types club @owenatgov 🏅

Second example works great. I'll explain by showing the JSDoc first:

```js
/**
 * @returns {string | undefined} Attribute value
 */
```

Since the compiler knows that calling **`$closestElement.getAttribute()`** returns `string | null` your first suggestion would need the JSDoc adjusting to be:

```js
/**
 * @returns {string | undefined | null} Attribute value
 */
```

Second suggestion solves that though 👍 

I've worked with teams in the past that don't like nested returns (often if/else) but I'll fit in with you. Some of the reasons why are better explained in the ESLint [`consistent-return`](https://eslint.org/docs/latest/rules/consistent-return) style rule (which StandardJS actually use)

Reply:
This was very educational, many thanks 😁 

I personally don't think this is a problem because we know that `return` on its own will return `undefined` which is in fact a value with meaning and it's a value we're happy with. I'm gonna wait and see what other team members say though.

Reply:
This has been [discussed more](https://github.com/alphagov/govuk-frontend/pull/2987/files#r1073800280) so I'll bring back early returns soon

Will prefer `null` vs `undefined` for functions that "return a selector query"